### PR TITLE
Use new tailwind plugin approach for hero icons

### PR DESCRIPTION
**Pull request by chrismccord**

### CODE REVIEW

**Review by jeregrine — APPROVED** [review]

File: installer/templates/phx_assets/tailwind.config.js [hunk 1]
```diff
@@ -18,9 +20,45 @@ module.exports = {
...
+      let valuesMap = new Map([
+        ["", "/24/outline"],
+        ["-solid", "/24/solid"],
+        ["-mini", "/20/solid"]
+      ].flatMap(([suffix, dir]) => readDir(dir).map(file => [file, suffix, dir]))
+       .map(([file, suffix, dir]) => {
+        let name = path.basename(file, ".svg") + suffix
+        let fullPath = path.join(iconsDir, dir, file)
+        return [name, {name, fullPath}]
+      }))
```
Comment (josevalim) [inline installer/templates/phx_assets/tailwind.config.js:48]:
```suggestion
      ].flatMap(([suffix, dir]) => readDir(dir).map(file => {
        let name = path.basename(file, ".svg") + suffix
        let fullPath = path.join(iconsDir, dir, file)
        return [name, {name, fullPath}]
      }))
```

File: installer/templates/phx_assets/tailwind.config.js [hunk 2]
```diff
@@ -18,9 +20,45 @@ module.exports = {
...
+            "mask": `var(--hero-${name})`,
+            "vertical-align": "middle",
+            "display": "inline-block"
+          }
+        }
+      }, {values: Object.fromEntries(valuesMap)})
+    })
   ]
-}
+}
```
Comment (josevalim) [inline installer/templates/phx_assets/tailwind.config.js:64]:
```suggestion
}

```

File: installer/templates/phx_assets/tailwind.config.js [hunk 3]
```diff
@@ -18,9 +20,48 @@ module.exports = {
...
     plugin(({addVariant}) => addVariant("phx-no-feedback", [".phx-no-feedback&", ".phx-no-feedback &"])),
     plugin(({addVariant}) => addVariant("phx-click-loading", [".phx-click-loading&", ".phx-click-loading &"])),
     plugin(({addVariant}) => addVariant("phx-submit-loading", [".phx-submit-loading&", ".phx-submit-loading &"])),
-    plugin(({addVariant}) => addVariant("phx-change-loading", [".phx-change-loading&", ".phx-change-loading &"]))
+    plugin(({addVariant}) => addVariant("phx-change-loading", [".phx-change-loading&", ".phx-change-loading &"])),
+
+    // Embeds Hero Icons (https://heroicons.com) into your app.css bundle
+    // See your `CoreComponents.icon/1` for more information.
+    //
+    plugin(function({matchComponents, theme}) {
```
Comment (josevalim) [inline installer/templates/phx_assets/tailwind.config.js:36]:
```suggestion
    plugin(function({matchComponents}) {
```

**Review by josevalim — APPROVED** [review]

**Comment by chrismccord** [comment]
❤️❤️❤️🐥🔥

**Pull request merged**